src/NeverDestroyed_TEST.cc

```diff
@@ -0,0 +1,95 @@
+#include <gtest/gtest.h>
+
+#include <random>
+#include <unordered_map>
+
+#include "ignition/utils/NeverDestroyed.hh"
+
+class Boom : public std::exception { };
+struct DtorGoesBoom {
+  ~DtorGoesBoom() noexcept(false) { throw Boom(); }
+};
+
+// Confirm that we see booms by default.
+GTEST_TEST(NeverDestroyed, BoomTest) {
+  try {
+    { DtorGoesBoom foo; }
+    GTEST_FAIL();
+  } catch (const Boom&) {
+    ASSERT_TRUE(true);
+  }
+}
+
+// Confirm that our wrapper stops the booms.
+GTEST_TEST(NeverDestroyed, NoBoomTest) {
+  try {
+    { ignition::utils::NeverDestroyed<DtorGoesBoom> foo; }
+    ASSERT_TRUE(true);
+  } catch (const Boom& e) {
+    GTEST_FAIL();
+  }
+}
+
+// This is an example from the class overview API docs; we repeat it here to
```

Comment:
Not what this is referring to. I don't see any examples in the API docs.

Reply:
I added the docs from Drake in, for completeness.  The reference should make sense now.